server/src/main.rs

```diff
@@ -217,3 +221,17 @@ fn check_if_table_or_model_table_is_initialized_otherwise_log_error<T>(
         true
     }
 }
+
+/// Register a handler to execute when CTRL+C is pressed. The handler takes an
+/// exclusive lock for the storage engine, flushes the data the storage engine
+/// currently buffers, and terminates the system without releasing the lock.
+fn setup_ctrl_c_handler(context: &Arc<Context>) {
+    let ctrl_c_context = context.clone();
+    context.runtime.spawn(async move {
+        // Errors are consciously ignored as the program should terminate if the
+        // handler cannot be registered as buffers otherwise cannot be flushed.
+        tokio::signal::ctrl_c().await.unwrap();
+        ctrl_c_context.storage_engine.write().unwrap().flush();
+        std::process::exit(0)
```

Comment:
Should we return a non-0 code when stopped by SIGINT?

Reply:
The code flushes all data currently in memory and exists. So if `std::process::exit(0)` is called the system have been terminated successfully. Also there does not seem to be a cross-platform consensus on what each error code means for the none-zero error codes. Thus, I think we should simply return zero.

Reply:
I agree